hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/OMClientRequest.java

```diff
@@ -168,11 +169,39 @@ public OzoneManagerProtocolProtos.UserInfo getUserInfo() throws IOException {
     OzoneManagerProtocolProtos.UserInfo.Builder userInfo =
         OzoneManagerProtocolProtos.UserInfo.newBuilder();
 
-    // If S3 Authentication is set, determine user based on access ID.
+    // If S3 Authentication is set, determine user based on STS token first,
+    // falling back to accessId if session token not present.
     if (omRequest.hasS3Authentication()) {
-      String principal = OzoneAclUtils.accessIdToUserPrincipal(
-          omRequest.getS3Authentication().getAccessId());
-      userInfo.setUserName(principal);
+      final String accessKeyId = omRequest.getS3Authentication().getAccessId();
+      if (accessKeyId.startsWith("ASIA") && !omRequest.getS3Authentication().hasSessionToken()) {
+        throw new IOException("Error with STS token", new AuthenticationException(
+            "Missing session token for accessKeyId: " + accessKeyId));
+      }
+      if (omRequest.getS3Authentication().hasSessionToken()) {
+        try {
+          final STSTokenIdentifier stsTokenIdentifier = OzoneManager.getStsTokenIdentifier();
+          if (stsTokenIdentifier != null) {
+            final String originalAccessKeyId = stsTokenIdentifier.getOriginalAccessKeyId();
```

Comment:
You might be able to reuse the static  `OzoneManager.getS3AuthEffectiveAccessId()` to reduce a bit of repeated code here? See if you think it makes sense?